extensions/remote-udp/README.md

```diff
@@ -0,0 +1,65 @@
+# Remote UDP Extension
+
+The openEO API is a specification for interoperable cloud-based processing of large Earth observation datasets.
+
+This extension enables user to load user-defined processes that are hosted on other hosts through the process namespace into process graphs.
+
+- Version: **0.1.0**
+- Stability: **experimental**
+- Conformance class: `https://api.openeo.org/extensions/remote-udp/0.1.0`
+
+## Justification
+
+The openEO API defines the `namespace` property in a process node of a process graph as follows:
+>  The following options are predefined by the openEO API, but additional namespaces may be introduced by back-ends or in a future version of the API.
+> * `null` [...]
+> * `backend` [...]
+> * `user` [...]
+
+This makes it possible that this extension adds additional allowed values to the `namespace` property.
+
+## Specification
+
+This extension extends the `namespace` property of process graph nodes so that it accepts **absolute** URL with the protocols `https` (**recommended**) and `http` (discouraged). The URLs specified MUST either return
+
+1. a single process (compatible to the endpoint `GET /process_graphs/{process_graph_id}`)
+2. a list of processes (compatible to the endpoint `GET /process_graphs`)
+
+If a URL is provided for the `namespace` property, the `id` property of the process graph node MUST be set to the `id` of the process.
+
+### Compatibility
+
+Compatible means in this context that the requests and responses must comply to the openEO API specification with the following exceptions:
+
+- The `Authorization` header is not required and will usually not be sent.
+- Lists of processes MUST NOT paginate and the full process description MUST be provided for each process (i.e., the recommendation to omit large properties such as `process_graph` doesn't apply).
+
+### Client Considerations
+
+Clients MUST only offer this functionality to users if the conformance class of this extension is listed in the `conformsTo` property of the `GET /` endpoint.
+
+The protocol `http` is discouraged for URLs as web-based clients may not be able to retrieve HTTP URLs from a HTTPS context.
+For the same reason it is also RECOMMENDED to enable CORS for all URLs.
```

Comment:
> RECOMMENDED to enable CORS

this is not very practical. In most cases the openEO user, client, nor backend will have control over the hosting solution.
I'm afraid web-based clients will have a hard time to resolve the URL if they want to inspect it. But I'm not sure this is a real problem as it's acctually the backend's task to  resolve the URL for evaluation

Reply:
But if you load such a process into the Web Editor it will not work well. As such I think a recommendation is fine.